bitcoin/src/bip32.rs

```diff
@@ -409,12 +606,12 @@ impl DerivationPath {
 
     /// Get an [Iterator] over the unhardened children of this [DerivationPath].
     pub fn normal_children(&self) -> DerivationPathIterator {
-        DerivationPathIterator::start_from(self, ChildNumber::Normal { index: 0 })
+        DerivationPathIterator::start_from(self, ChildNumber::Normal(NormalChildNumber(0)))
     }
 
     /// Get an [Iterator] over the hardened children of this [DerivationPath].
     pub fn hardened_children(&self) -> DerivationPathIterator {
-        DerivationPathIterator::start_from(self, ChildNumber::Hardened { index: 0 })
+        DerivationPathIterator::start_from(self, ChildNumber::Hardened(HardenedChildNumber(0)))
```

Comment:
Maybe changing this to the constants `Number::ZERO`

Reply:
You mean `ZERO_HARDENED` as in? https://github.com/rust-bitcoin/rust-bitcoin/blob/7387e9c6b8f4857894eaca540fe1521069452f13/bitcoin/src/bip32.rs#L172-L173

Reply:
The type is already called type of the number. 
For the user will be  `NormalChildNumber::ZERO`

Reply:
```suggestion
        DerivationPathIterator::start_from(self, ZERO_NORMAL)
    }

    /// Get an [Iterator] over the hardened children of this [DerivationPath].
    pub fn hardened_children(&self) -> DerivationPathIterator {
        DerivationPathIterator::start_from(self, ZERO_HARDENED)
```

These are equivalent no? What am I missing here?

Reply:
Yes, fixed.